src/Dropdown/index.jsx

```diff
@@ -1,225 +1,169 @@
-      this.setState({
-        focusIndex: (this.state.focusIndex + 1) % this.props.menuItems.length,
-      });
-    } else if (Dropdown.isTriggerKey('NAVIGATE_UP', e.key)) {
-      this.setState({
-        focusIndex: ((this.state.focusIndex - 1) + this.props.menuItems.length) %
-                    this.props.menuItems.length,
-      });
     }
   }
 
-  handleToggleKeyDown(e) {
-    if (!this.state.open && Dropdown.isTriggerKey('OPEN_MENU', e.key)) {
-      this.toggle();
-    } else if (this.state.open && Dropdown.isTriggerKey('CLOSE_MENU', e.key)) {
-      this.toggle();
+  handleMenuKeyDown = (e) => {
+    switch (e.key) {
+      case 'ArrowUp':
+        e.preventDefault();
+        this.focusPrevious();
+        break;
+      case 'ArrowDown' || 'Tab':
+        e.preventDefault();
+        this.focusNext();
+        break;
+      case 'Escape':
+        e.stopPropagation();
+        this.toggle();
+        break;
+      default:
+        return null;
     }
   }
 
-  toggle() {
+  open = () => {
+    // adding event listener here so the user can close dropdown on click outside of the dropdown
+    document.addEventListener('click', this.handleDocumentClick, true);
     this.setState({
-      open: !this.state.open,
-      focusIndex: 0,
+      open: true,
     });
   }
 
-  generateMenuItems(menuItems) {
-    return menuItems.map((menuItem, i) => {
-      if (React.isValidElement(menuItem)) {
-        const cloneProps = {
-          ref: (item) => { this.menuItems[i] = item; },
-          className: 'dropdown-item',
-          key: i,
-          onKeyDown: this.handleMenuKeyDown,
-        };
-        return React.cloneElement(menuItem, cloneProps);
-      }
-      return (
-        <a
-          className="dropdown-item"
-          href={menuItem.href}
-          key={menuItem.href}
-          onKeyDown={this.handleMenuKeyDown}
-          ref={(item) => {
-            this.menuItems[i] = item;
-          }}
-          role="menuitem"
-        >
-          {menuItem.label}
-        </a>
-      );
+  close = () => {
+    document.removeEventListener('click', this.handleDocumentClick, true);
+    this.setState({
+      open: false,
     });
   }
 
+  toggle = () => {
+    if (this.state.open) {
+      this.close();
+    } else {
+      this.open();
+    }
+  }
+
   render() {
     const { open } = this.state;
     const {
-      buttonType,
-      iconElement,
-      menuItems,
-      title,
+      buttonClassName,
+      buttonContent,
+      ...other
     } = this.props;
-    const hasIconElement = React.isValidElement(iconElement);
 
     return (
       <div
-        className={classNames(
-          'dropdown',
-          {
-            show: open,
-            'has-icon': hasIconElement,
-            rounded: hasIconElement,
-            border: hasIconElement,
-            'd-flex': hasIconElement,
-            'bg-white': hasIconElement,
-          },
-        )}
+        className={this.state.open ? 'dropdown show' : 'dropdown'}
```

Comment:
Let's be consistent on referencing `open`. Here it's `this.state.open` but below it's `open`

Reply:
Removed `const { open } = this.state;` to keep clear/consistent as `this.state.open`